fuel-vm/src/tests/predicate.rs

```diff
@@ -563,3 +563,48 @@ async fn gas_used_by_predicates_more_than_limit() {
         CheckError::PredicateVerificationFailed(_)
     ));
 }
+
+#[test]
+#[ntest::timeout(5_000)]
+fn synchronous_estimate_predicates_respects_total_tx_gas_limit() {
+    let limit = 1_000_000;
+    let rng = &mut StdRng::seed_from_u64(2322u64);
+
+    let params = CheckPredicateParams {
+        max_gas_per_predicate: limit,
+        max_gas_per_tx: limit,
+        gas_costs: GasCosts::unit(),
+        ..Default::default()
+    };
+
+    // Infinite loop
+    let predicate = vec![op::noop(), op::jmpb(RegId::ZERO, 0)]
+        .into_iter()
+        .collect::<Vec<u8>>();
+    let predicate_owner = Input::predicate_owner(&predicate);
+
+    let mut builder = TransactionBuilder::script(vec![], vec![]);
+    builder.max_fee_limit(1000).maturity(Default::default());
+
+    let coin_amount = 100_000;
+
+    for _ in 0..255 {
+        builder.add_input(Input::coin_predicate(
+            rng.gen(),
+            predicate_owner,
+            coin_amount,
+            AssetId::default(),
+            rng.gen(),
+            0,
+            predicate.clone(),
+            vec![],
+        ));
+    }
+    let mut transaction = builder.finalize();
+
+    // When
+    let result = transaction.estimate_predicates(&params, MemoryInstance::new());
+
+    // Then
+    assert_eq!(Ok(()), result);
```

Comment:
if the estimation runs out of gas due to each predicate running an infinite loop, why is this an `Ok(())`?

Reply:
It was done a long time ago to allow estimate predicates even if they return false. For cases when signature incorrect and predicate fails

I also don't like it=D I was thinking of modifying the code and returning an error in all cases except false predicate. But it will change how estimation works, so I decided not to include it in this PR.

Reply:
Ok - this seems odd but makes sense given the context